extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/taskadapter/PodTemplateTaskAdapter.java

```diff
@@ -130,15 +112,7 @@ public PodTemplateTaskAdapter(
   @Override
   public Job fromTask(Task task) throws IOException
   {
-    PodTemplateSelectStrategy podTemplateSelectStrategy;
-    KubernetesTaskRunnerDynamicConfig dynamicConfig = dynamicConfigRef.get();
-    if (dynamicConfig == null || dynamicConfig.getPodTemplateSelectStrategy() == null) {
-      podTemplateSelectStrategy = KubernetesTaskRunnerDynamicConfig.DEFAULT_STRATEGY;
-    } else {
-      podTemplateSelectStrategy = dynamicConfig.getPodTemplateSelectStrategy();
-    }
-
-    PodTemplateWithName podTemplateWithName = podTemplateSelectStrategy.getPodTemplateForTask(task, templates);
+    PodTemplateWithName podTemplateWithName = podTemplateSelector.getPodTemplateForTask(task).get();
```

Comment:
Since the interface returns an Optional - this should handle an `Optional.empty` response from the selector